File: java/05-reactor/src/test/java/com/fResult/reactor/ch5_05/MapTest.java
package com.fResult.reactor.ch5_05;

import lombok.val;
import org.junit.jupiter.api.Test;
import reactor.core.publisher.Flux;
import reactor.test.StepVerifier;

class MapTest {
  @Test
  void maps() {
    val characters = Flux.just("a", "b", "c").map(String::toUpperCase);

    StepVerifier.create(characters).expectNext("A", "B", "C").verifyComplete();
  }
}

File: kotlin/05-reactor/src/test/kotlin/com/fResult/reactor/ch5_05/MapTest.kt
package com.fResult.reactor.ch5_05

import reactor.core.publisher.Flux
import reactor.test.StepVerifier
import kotlin.test.Test

class MapTest {
  @Test
  fun maps() {
    val upperStrings = Flux.just("a", "b", "c", "d", "e").map(String::uppercase)
    StepVerifier.create(upperStrings).expectNext("A", "B", "C", "D", "E").verifyComplete()
  }
}
